### PR TITLE
FontDisplay: Fixing missed equals and hashcode overrides

### PR DESCRIPTION
**Pull request by Windchild292**
Missed these in the initial commit. I also simplified the constructor, which helps with setting the value nicely.

### CODE REVIEW

File: megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java [hunk 1]
```diff
@@ -35,6 +37,10 @@
     //endregion Variable Declarations
 
     //region Constructors
+    public FontDisplay(final String familyName) {
+        this(Font.decode(familyName).deriveFont(Font.PLAIN, 16f), familyName);
```
Comment (SJuliez) [inline megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java:41]: If you're interested, you could try getting the UIManager default font for comboboxes and derive from it. The size would probably be 12 then.
Like:
Font font = UIManager.getFont("ComboBox.font");
if (font != null) etc.

image

File: megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java [hunk 2]
```diff
@@ -49,12 +55,34 @@ public Font getFont() {
...
+    @Override
+    public boolean equals(final @Nullable Object other) {
+        if (other == null) {
+            return false;
+        } else if (this == other) {
+            return true;
+        } else if (other instanceof FontDisplay) {
+            return getFont().equals(((FontDisplay) other).getFont());
+        } else if (other instanceof Font) {
+            return getFont().equals(other);
```
Comment (SJuliez) [inline megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java:76]: As someone who did not write this class I would never write fontDisplay.equals("Verdana") or fontDisplay.equals(someFont) and expect it to return true without taking a deep look at it first. Far be it from me to create unnecessary work but in this case I'd say isFont(String fontName) and isFont(Font font) would be easier on other users of this class such as me.

**Review by SJuliez — CHANGES_REQUESTED** [review]
Sorry, small PR only and still I'd like to request a change to the equals...

File: megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java]: Both were testing setups that don't compare properly (which lead to the constructor change), so I've removed them. I was trying to simplify selected item assignment by making them possible.

File: megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/displayWrappers/FontDisplay.java]: It's in a constructor, so it needs to be inline... which means using the UIManager isn't the most stable way to code. However, I've swapped to 12 as that's the default.

**Review by Windchild292 — COMMENTED** [review]
I concur on both, review changes applied

**Review by SJuliez — APPROVED** [review]

**Pull request merged**